flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosApplicationMasterRunner.java

```diff
@@ -0,0 +1,618 @@
+
+			// try to start the actor system, JobManager and JobManager actor system
+			// using the configured address and ports
+			actorSystem = BootstrapTools.startActorSystem(config, appMasterHostname, listeningPort, LOG);
+
+			final String akkaHostname = AkkaUtils.getAddress(actorSystem).host().get();
+			final int akkaPort = (Integer) AkkaUtils.getAddress(actorSystem).port().get();
+
+			LOG.info("Actor system bound to hostname {}.", akkaHostname);
+
+			// try to start the artifact server
+			LOG.debug("Starting Artifact Server");
+			final int artifactServerPort = config.getInteger(ConfigConstants.MESOS_ARTIFACT_SERVER_PORT_KEY,
+				ConfigConstants.DEFAULT_MESOS_ARTIFACT_SERVER_PORT);
+			artifactServer = new MesosArtifactServer(sessionID, akkaHostname, artifactServerPort);
+
+			// ----------------- (3) Generate the configuration for the TaskManagers -------------------
+
+			final Configuration taskManagerConfig = BootstrapTools.generateTaskManagerConfiguration(
+				config, akkaHostname, akkaPort, slotsPerTaskManager, TASKMANAGER_REGISTRATION_TIMEOUT);
+			LOG.debug("TaskManager configuration: {}", taskManagerConfig);
+
+			final Protos.TaskInfo.Builder taskManagerContext = createTaskManagerContext(
+				config, mesosConfig, ENV,
+				taskManagerParameters, taskManagerConfig,
+				workingDir, getTaskManagerClass(), artifactServer, LOG);
+
+			// ----------------- (4) start the actors -------------------
+
+			// 1) JobManager & Archive (in non-HA case, the leader service takes this)
+			// 2) Web Monitor (we need its port to register)
+			// 3) Resource Master for Mesos
+			// 4) Process reapers for the JobManager and Resource Master
+
+			// 1: the JobManager
+			LOG.debug("Starting JobManager actor");
+
+			// we start the JobManager with its standard name
+			ActorRef jobManager = JobManager.startJobManagerActors(
+				config, actorSystem,
+				new scala.Some<>(JobManager.JOB_MANAGER_NAME()),
+				scala.Option.<String>empty(),
+				getJobManagerClass(),
+				getArchivistClass())._1();
+
+
+			// 2: the web monitor
+			LOG.debug("Starting Web Frontend");
+
+			webMonitor = BootstrapTools.startWebMonitorIfConfigured(config, actorSystem, jobManager, LOG);
+			if(webMonitor != null) {
+				final URL webMonitorURL = new URL("http", appMasterHostname, webMonitor.getServerPort(), "/");
+				mesosConfig.frameworkInfo().setWebuiUrl(webMonitorURL.toExternalForm());
+			}
+
+			// 3: Flink's Mesos ResourceManager
+			LOG.debug("Starting Mesos Flink Resource Manager");
+
+			// create the worker store to persist task information across restarts
+			MesosWorkerStore workerStore = createWorkerStore(config);
+
+			// we need the leader retrieval service here to be informed of new
+			// leader session IDs, even though there can be only one leader ever
+			LeaderRetrievalService leaderRetriever =
+				LeaderRetrievalUtils.createLeaderRetrievalService(config, jobManager);
+
+			Props resourceMasterProps = MesosFlinkResourceManager.createActorProps(
```

Comment:
Resource master or resource manager? I think it should be `resourceManagerProps`.